discovery_engine_core/web-api/openapi.yaml

```diff
@@ -127,6 +127,45 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/IngestionError'
+    delete:
+      tags:
+        - documents
+      summary: Deletes all listed documents.
+      description: |-
+        Deletes all documents listed in the request body. The endpoint is
+        idempotent. I.e. If the list contains one or multiple non-existing
+        document it does not produce  an error.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/DeleteDocumentsRequest'
+      responses:
+        '204':
+          description: successful operation
+        '400':
+          description: invalid request
```

Comment:
❔ In the ticket we say that we that in case of an error we return the id of the documents that could not be deleted. Has this requirement changed?

Reply:
Good point but the endpoint is also idempotent so:

- this doesn't apply if the document can't be deleted because it is already deleted
- it also doesn't really apply for general "everything went wrong and we deleted nothing" errors
- so it only applies if while we delete documents somewhere in between it failed and some
  documents are deleted and some not and we _know_ that some are deleted
  - but the deletion in postgres is transactional so this can't happen there
  - and the deletion in elastic search could (and should) be batched, too
  
So isn't it a situation where it always either returns all or none documents and in the exceedingly 
rare case where this isn't the case (we din't implement batched ec deletion and internet/ec crashed in between)
it would be okay to tell the user to just redo the full deletion due to idempotency.

Reply:
daily: we don't need to return ids but deletion at elastic search must use batch mode